internal/tar_interpreter.go

```diff
@@ -45,42 +45,56 @@ func (tarInterpreter *FileTarInterpreter) unwrapRegularFile(fileReader io.Reader
 		}
 	}
 	fileDescription, haveFileDescription := tarInterpreter.Sentinel.Files[fileInfo.Name]
+	isIncremented := haveFileDescription && fileDescription.IsIncremented
+	// todo: clearer catchup backup detection logic
+	isCatchup := tarInterpreter.createNewIncrementalFiles
 
-	// If this file is incremental we use it's base version from incremental path
-	if haveFileDescription && tarInterpreter.Sentinel.IsIncremental() && fileDescription.IsIncremented {
-		err := ApplyFileIncrement(targetPath, fileReader, tarInterpreter.createNewIncrementalFiles)
-		return errors.Wrapf(err, "Interpret: failed to apply increment for '%s'", targetPath)
+	if localFileInfo, _ := getLocalFileInfo(targetPath); localFileInfo != nil {
+		isPageFile := isPagedFile(localFileInfo, targetPath)
```

Comment:
You can check it inside of unwrapToExistFile and reduce number of arguments.

Reply:
In this case I'll need to need to pass the localFileInfo to unwrapToExistFile so the arguments count won't change